app/models/ability.rb

```diff
@@ -28,11 +28,14 @@ def initialize(user)
     #
     # See the wiki for details:
     # https://github.com/ryanb/cancan/wiki/Defining-Abilities
-    user ||= User.new # guest user (not logged in)
-    if user.role == "superadmin"
-      can :manage, :all
-    elsif user.role == "admin"
-      can :read, :all
+
+    user.permissions.each do |permission|
+      permission_class = (permission.subject_class.match(/:?all/i)) ?
```

Comment:
Regex is helpful in this case because all, :all, and ALL would all be valid strings. I think it makes the code a bit more robust.